data-prepper-pipeline-parser/build.gradle

```diff
@@ -12,7 +12,23 @@ group = 'org.opensearch.dataprepper.core'
 dependencies {
     implementation project(':data-prepper-api')
     implementation project(':data-prepper-plugins:blocking-buffer')
-    implementation 'com.fasterxml.jackson.core:jackson-databind'
+    implementation 'com.fasterxml.jackson.core:jackson-databind:2.12.3'
     implementation 'com.fasterxml.jackson.dataformat:jackson-dataformat-yaml'
     implementation 'org.apache.commons:commons-collections4:4.4'
+    implementation 'org.apache.commons:commons-jexl3:3.2.1'
+    implementation 'org.projectlombok:lombok:1.18.22'
+    implementation 'com.jayway.jsonpath:json-path:2.6.0'
+    implementation 'javax.inject:javax.inject:1'
+    implementation(libs.spring.core) {
+        exclude group: 'commons-logging', module: 'commons-logging'
+    }
+    implementation(libs.spring.context) {
+        exclude group: 'commons-logging', module: 'commons-logging'
+    }
+    implementation 'software.amazon.cloudwatchlogs:aws-embedded-metrics:2.0.0-beta-1'
+    testImplementation 'org.apache.logging.log4j:log4j-jpl:2.23.0'
+    testImplementation 'org.assertj:assertj-core:3.20.2'
```

Comment:
We should be able to remove this. There is one import to change to Hamcrest and then we can remove.

Reply:
hamcrest does not seem to support usingRecursiveComparison.